src/Native/Unix/System.Native/pal_process.c

```diff
@@ -339,7 +339,21 @@ int32_t SystemNative_ForkAndExecProcess(const char* filename,
     // on another thread while a vfork() child is still pending, bad things are possible; however we
     // do not do that.
 
+#if defined (__GLIBC__)
     if ((processId = vfork()) == 0) // processId == 0 if this is child process
+#else
+    // musl libc has an undocumented failure mode around setuid(); we must exclude it.
+    if (setCredentials)
+    {
+        processId = fork();
```

Comment:
Since the vfork is an optimization, and setCredentials not a likely case, we could behave the same for __GLIBC__.

Reply:
I think that makes sense. @janvorli?

Reply:
Please add servicing-consider label back when tihs is resolved.